ports/espressif/.gitignore

```diff
@@ -1,2 +1,2 @@
-build*/
-sdkconfig.old
+# lock files for examples and components
+dependencies.lock
```

Comment:
Did you mean to remove `build*/` and `sdkconfig.old`?

Reply:
I haven't noticed those appearing with build commands mentioned in `Makefile`.

Reply:
I have `build-*` directories:
```sh
~/repos/circuitpython/ports/espressif$ ls -d build*
build-adafruit_feather_esp32s2  build-adafruit_feather_esp32s2_tft  build-adafruit_feather_esp32_v2  build-adafruit_qtpy_esp32_pico
```
`sdkconfig.old` files get created by running `idf.py menuconfig`

Reply:
Following are excluded in the CP root `.gitignore`.
```
build/
build-*/
```

Reply:
I'll readd `sdkconfig`.

Reply:
what creates dependencies.lock?

Reply:
> what creates dependencies.lock?

It's the [IDF Component Manager](https://docs.espressif.com/projects/esp-idf/en/latest/api-guides/tools/idf-component-manager.html).

Reply:
Thanks, didn't know `build*` was in the root `.gitignore`. It's the only entry in a bunch of `ports/*/.gitignore` files.

Reply:
I will add some commits to this PR to remove the redundant `port/` .gitignore's, unless you are already on it.

Reply:
> I will add some commits to this PR to remove the redundant `port/` .gitignore's...

Sure👍